src/stations.rs

```diff
@@ -26,11 +26,57 @@ impl StationCatalogue {
 
     /// Find the station nearest to the given coordinates.
     pub fn find_near(&self, location: &model::Coordinates) -> &Station {
-        unimplemented!()
+        // use std::iter::Iterator;
+        use crate::compute::gcd::great_circle_distance;
+        let cmp_distance = |s1: &&Station, s2: &&Station| {
+            let d1 = great_circle_distance(&s1.coordinates, location);
+            let d2 = great_circle_distance(&s2.coordinates, location);
+            d1.partial_cmp(&d2).expect("Distances shouldn't be NaN")
+        };
+        self.stations
+            .iter()
+            .min_by(cmp_distance)
+            .expect("StationCatalogue has at least one station, so there must be a minimum")
     }
 
     /// Add a station's data to this catalogue, assigning it an appropriate unique id.
     fn add(&mut self, name: &str, coords: &model::Coordinates) {
```

Comment:
I feel like we should try to standardize on `coords`, `coordinates` or `location` for these arguments/field names. Right now we're using all three in this one file, which seems like a little bit much.

Reply:
I'd prefer `coordinates`,  on the basis that I would understand a location to be a named place or otherwise have some meaning beyond "this point on the globe".

Reply:
sounds good to me. It also matches the type name directly, which seems reasonable.